### PR TITLE
Order History Page Skeleton

### PR DESCRIPTION
**Pull request by supernova-at**
## Description

This PR introduces a skeleton 🦴  for the Order History Page.

It:

- Creates an Order History page
- Links to it from the Account Menu
- Redirects the user to the home page if they are signed out
- Updates the browser tab title
- Displays a full page loading indicator while the page is loading data
- Displays a message if the user has no order history items (dummy data)

## Related Issue

Closes PWA-276.

### Verification Steps

1. Sign In
1. Go to the Account Menu links and click "Order History"
1. Verify that you land on the Order History page
1. Verify that the browser tab title is "Order History - Venia"
1. Sign out
1. Verify that you are redirected to the home page

## Checklist

- [x] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2611.pwa-venia.com : LH Performance Expected 0.85 Actual 0.34, LH Accessibility Expected 1 Actual 0.97, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 90 Actual 39.333333333333
https://pr-2611.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.34, LH Best Practices Expected 1 Actual 0.92
https://pr-2611.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.39, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 65 Actual 60

**Comment by supernova-at** [comment]
UX Feedback: The Account Menu popup / left navigation should close when the link is clicked. When opened again, it should show the account menu links (not the main menu)

**Comment by brendanfalkowski** [comment]
@supernova-at Would you be interested in a PR for an `AuthRoute` component that's used like this:

Screen Shot 2020-08-06 at 11 49 47 AM

It automatically bounces signed out users to our route for signing in, so you don't have to manage this logic in every auth-only area of the site.

File: packages/peregrine/lib/talons/Header/useAccountTrigger.js [hunk 1]
```diff
@@ -55,9 +55,11 @@ export const useAccountTrigger = props => {
     }, [setAccountMenuIsOpen]);
 
     // Close the Account Menu on page change.
+    // Trigger on location.key to close even when the page "changes" to the current page.
+    // This can happen when clicking on a link to a page you're already on, for example.
     useEffect(() => {
         setAccountMenuIsOpen(false);
-    }, [location.pathname, setAccountMenuIsOpen]);
+    }, [location.key, setAccountMenuIsOpen]);
```
Comment (supernova-at) [inline packages/peregrine/lib/talons/Header/useAccountTrigger.js:62]: Double check that we can't just use `location` here.

File: packages/venia-ui/lib/components/AccountMenu/accountMenuItems.js [hunk 2]
```diff
@@ -39,7 +34,7 @@ const AccountMenuItems = props => {
             {menuItems}
             <button
                 className={classes.signOut}
-                onClick={handleSignOut}
+                onClick={onSignOut}
```
Comment (supernova-at) [inline packages/venia-ui/lib/components/AccountMenu/accountMenuItems.js:37]: Consider creating a `handleSignOut` here.

**Comment by supernova-at** [comment]
> Displays a full page loading indicator while the page is loading data

Use the page loading indicator for this.

File: packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const EMPTY_DATA_MESSAGE = `You don't have any orders yet.`;
+
+const OrderHistoryPage = props => {
+    const talonProps = useOrderHistoryPage();
+    const { data } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    let pageContents;
+    if (!data) {
```
Comment (sirugh) [inline packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js:19]: Let's get ahead of `cache-and-network` and always use `loading && !data` to only render empty/loading states when we're loading and have nothing else to render.

**Comment by dpatil-magento** [comment]
@supernova-at Please check observations - 
1. Communications link is navigating to Home page instead of Communications.
2. After click Order History from left drawer > Left drawer does not get closed automatically.

**Comment by dpatil-magento** [comment]
> @supernova-at Please check observations -
> 
> 1. Communications link is navigating to Home page instead of Communications.
> 2. After click Order History from left drawer > Left drawer does not get closed automatically.

It was caused by stale deployment. Looks good, QA approved.

**Pull request merged**